src/tools/fuzzing/fuzzing.cpp

```diff
@@ -1067,12 +1074,21 @@ Expression* TranslateToFuzzReader::makeImportLogging() {
 }
 
 Expression* TranslateToFuzzReader::makeImportThrowing(Type type) {
+  // TODO: This and makeThrow should probably be rare, as they halt the program.
+
   // We throw from the import, so this call appears to be none and not
   // unreachable.
   assert(type == Type::none);
 
-  // TODO: This and makeThrow should probably be rare, as they halt the program.
-  return builder.makeCall(throwImportName, {}, Type::none);
+  // An argument of 0 means to throw a JS exception, and otherwise the value in
+  // a wasm tag. Emit 0 or non-zero with ~equal probability.
+  Expression* arg;
+  if (oneIn(2)) {
+    arg = builder.makeConst(int32_t(0));
+  } else {
+    arg = makeConst(Type::i32);
+  }
+  return builder.makeCall(throwImportName, {arg}, Type::none);
```

Comment:
See the comment in `callImport`

Reply:
If in case we want to throw an empty exception, can't we pass `{}` instead of `{arg}`?

Reply:
The signature of the function is
```wat
(import "fuzzing-support" "throw" (func $throw (param i32)))
```
so we have to put something there.

We could have two separate functions, "throw JS" and "throw Wasm" perhaps? Then the JS one would have signature
```wat
(import "fuzzing-support" "throw" (func $throw))
```

Reply:
I thought this `makeCall` somehow calls `LoggingExternalInterface::callImport` and there we decide whether to actually call the `throw` from JS or just throw an empty exception, so I didn't think this line has to comply with the JS import API... But I guess I'm mistaken? This call does not always go to `LoggingExternalInterface::callImport`?

Reply:
This `builder.makeCall` makes a call in Binaryen IR. We then write it to a wasm file, which is then executed in several ways. One of the ways is in the Binrayen interpreter, `LoggingExternalInterface::callImport` as you said, but we also need the IR to be valid so that the output wasm file is valid, and so that the wasm can also run in `d8` for example.